src/main/java/uk/gov/hmcts/darts/common/service/impl/DataAnonymisationServiceImpl.java

```diff
@@ -48,13 +48,19 @@ public class DataAnonymisationServiceImpl implements DataAnonymisationService {
 
     private final CurrentTimeHelper currentTimeHelper;
     private final ExternalOutboundDataStoreDeleter outboundDataStoreDeleter;
-    private final CaseRepository caseRepository;
     private final TransformedMediaRepository transformedMediaRepository;
     private final TransientObjectDirectoryRepository transientObjectDirectoryRepository;
     private final LogApi logApi;
+    private final CaseService caseService;
+
 
     @Override
-    public void anonymizeCourtCaseEntity(UserAccountEntity userAccount, CourtCaseEntity courtCase) {
+    @Transactional
+    public void anonymizeCourtCaseById(Integer courtCaseId) {
```

Comment:
Optional - this method could be removed and instead call `anonymizeCourtCaseEntity(UserAccountEntity userAccount, CourtCaseEntity courtCase)` directly from `CaseExpiryDeletionAutomatedTask.java`, and call `getUserAccount()` outside of the for-loop so that the user is only retrieved once, as the call to getUserAccount() results in a database hit.

Reply:
Agreed and updated :)